lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -367,7 +367,7 @@ def track_lines(
     return refine_tracks_centroid(kymotrack_group, track_width=line_width, bias_correction=True)
 
 
-def filter_tracks(tracks, minimum_length):
+def filter_tracks(tracks, minimum_length, *, minimum_duration=0):
```

Comment:
this seems a bit odd to me to enforce a minimum number of frames on top of a minimum duration. Filtering by duration seems more natural and probably what you most often want, but how it's set up here you necessarily also still need to think about the number of tracked frames.

I could see multiple options to remedy this and give enough flexibility for users:
- add a default for `minimum_length=1`. This way you can do `filter_tracks(minimum_duration=x)` and it would work as you expect. You could also then choose to filter by both by doing `filter_tracks(n, minimum_duration=x)`
- deprecate this function and have 2 separate functions `filter_tracks_by_frames()` and `filter_tracks_by_duration` that you could then chain together if you want to do both

what do you think?

Reply:
Yeah, that's a good point.

I agree that having to always think about both is not nice. I'm just slightly concerned about polluting the namespace with 3 very similarly named functions (2 in the future then).

Personally, I would have liked for both arguments to just be keyworded, so you can see immediately from the calling code what is being used. But unfortunately, that'd be a breaking change.

We could just default `minimum_length` and warn that it will be keyworded in the future.

Reply:
yeah, I think I prefer the option to default both arguments. I think it's fine to warn that `minimum_length` will be keyword only in the future ... might be kinda annoying, but I guess that's the point 😉